lib/Dialect/Secret/Transforms/DistributeGeneric.cpp

```diff
@@ -684,13 +685,44 @@ void moveDialectAttrsToFuncArgument(Operation *top) {
 void moveMgmtAttrAnnotationFromInnerToOuter(Operation *top) {
   top->walk([&](secret::GenericOp genericOp) {
     auto *innerOp = &genericOp.getBody()->front();
-    auto mgmtAttr = innerOp->removeAttr(mgmt::MgmtDialect::kArgMgmtAttrName);
-    if (mgmtAttr) {
-      // only support a single result on the inner op
-      assert(genericOp->getNumResults() == 1 &&
-             "Only supporting single-result inner ops; file an issue if you "
-             "need multi-result ops here!");
-      genericOp.setResultAttr(0, mgmt::MgmtDialect::kArgMgmtAttrName, mgmtAttr);
+    auto yieldOp = genericOp.getYieldOp();
+    Attribute attr = innerOp->removeAttr(mgmt::MgmtDialect::kArgMgmtAttrName);
+    if (!attr) {
+      return;
+    }
+
+    if (yieldOp.getNumOperands() == 1) {
+      // If the yield op has only one operand, then it is the only result of
+      // the inner op, so we can just set the mgmt attr on the generic op.
+      genericOp.setResultAttr(0, mgmt::MgmtDialect::kArgMgmtAttrName, attr);
+      return;
+    }
+
+    for (OpOperand &yieldOperand : yieldOp->getOpOperands()) {
+      // The yield operand is produced by the inner op result, but may not
+      // correspond exactly to the quantity and order of results.
+      //
+      // E.g.,
+      //
+      //   %0, %1 = op.foo ... {mgmt.mgmt = [attr1, attr2]} : i16, i16
+      //   secret.yield %1, %0 : i16, i16
+      //
+      // Or
+      //
+      //   %0 = op.foo ... {mgmt.mgmt = attr} : i16
+      //   secret.yield %0, %0 : i16, i16
+      //
+      const auto &innerOpResult = cast<OpResult>(yieldOperand.get());
```

Comment:
I wonder what would happen if yieldOperand is a BlockArgument and could not be casted to OpResult.

Reply:
I believe it would crash, but this pass expects canonicalized to be run first which I think should fold such use away?